flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/index/FileDataIndexCacheTest.java

```diff
@@ -34,41 +35,41 @@
 import static org.apache.flink.runtime.io.network.partition.hybrid.HybridShuffleTestUtils.createAllUnreleasedRegions;
 import static org.assertj.core.api.Assertions.assertThat;
 
-/** Tests for {@link HsFileDataIndexCache}. */
-class HsFileDataIndexCacheTest {
-    private HsFileDataIndexCache indexCache;
+/** Tests for {@link FileDataIndexCache}. */
+class FileDataIndexCacheTest {
+    private FileDataIndexCache<HsFileDataIndexImpl.InternalRegion> indexCache;
 
-    private TestingFileDataIndexSpilledRegionManager spilledRegionManager;
+    private TestingFileDataIndexSpilledRegionManager<HsFileDataIndexImpl.InternalRegion>
+            spilledRegionManager;
 
     private final int numSubpartitions = 1;
 
-    private static final int SPILLED_INDEX_SEGMENT_SIZE = 256;
-
     private int numRetainedIndexEntry = 10;
 
     @BeforeEach
     void before(@TempDir Path tmpPath) throws Exception {
         Path indexFilePath = Files.createFile(tmpPath.resolve(UUID.randomUUID().toString()));
+        TestingFileDataIndexSpilledRegionManager.Factory<HsFileDataIndexImpl.InternalRegion>
+                testingSpilledRegionManagerFactory =
+                        new TestingFileDataIndexSpilledRegionManager.Factory<>();
         indexCache =
-                new HsFileDataIndexCache(
+                new FileDataIndexCache<>(
                         numSubpartitions,
                         indexFilePath,
                         numRetainedIndexEntry,
-                        TestingFileDataIndexSpilledRegionManager.Factory.INSTANCE);
-        spilledRegionManager =
-                TestingFileDataIndexSpilledRegionManager.Factory.INSTANCE
-                        .getLastSpilledRegionManager();
+                        testingSpilledRegionManagerFactory);
+        spilledRegionManager = testingSpilledRegionManagerFactory.getLastSpilledRegionManager();
     }
 
     @Test
     void testPutAndGet() {
         indexCache.put(0, createAllUnreleasedRegions(0, 0L, 3, 1));
-        Optional<InternalRegion> regionOpt = indexCache.get(0, 0);
+        Optional<HsFileDataIndexImpl.InternalRegion> regionOpt = indexCache.get(0, 0);
```

Comment:
It's a little confused that we test `HsFileDataIndexImpl. InternalRegion` for `FileDataIndexCacheTest`.

Reply:
I have add `TestingFileDataIndexRegion` to replace the `HsFileDataIndexImpl.InternalRegion` in the tests `FileDataIndexCacheTest` and `FileDataIndexSpilledRegionManagerImplTest`.